### DISCUSSION TITLE
API to replace text/bulk insert

### DISCUSSION THREAD

**Discussion by danielo515** [MunifTanjim/nui.nvim#124] [Ideas]
Hello again.
I've been using recently the Text component. It is great, but it has some limitations when you want to insert new text into new buffers.
The first thing that impressed me is that, the neovim API to insert text fails when you provide a length and there is no enough text to replace it. So what you should do instead is to provide the same value for both start and end. Because of this, the default values the Text components uses (which is setting the end to start + length) are not very suitable to insert new text.

After realising that I had to do cleanup before I want to insert new text led me to the conclusion that the better API for my usecase is to provide a buffer and replace it's text entirely. I'm not sure if I can concatenate the Text content somehow or if this use-case is not possible with the current API.

**Comment by MunifTanjim** [comment]
I'm not exactly sure how your code looks like. Can you post some codes showing a small example of what you're trying to do?

I think you can use `NuiLine` to render a line on the buffer, and `:append()` new texts to that line.

**Reply by danielo515** [reply]
Look exactly what I need.
I don't know how I overlooked it. I will give it a try, thanks
